akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpMessage.scala

```diff
@@ -268,8 +275,13 @@ final class HttpRequest(
   val uri:      Uri,
   val headers:  immutable.Seq[HttpHeader],
   val entity:   RequestEntity,
-  val protocol: HttpProtocol)
-  extends jm.HttpRequest with HttpMessage {
+  val protocol: HttpProtocol,
+  val stringRepresentation: HttpRequest ⇒ String = { request ⇒
```

Comment:
It looks a bit strange to have a function from `HttpRequest => String` on a `HttpRequest`. I understand how we got here: calculating the string representation in the constructor seems wasteful, but you want it to be customizable and be able to refer to it from the `toString`. Using a `lazy val` would be neat, but introduces 2 new fields to this class, while adding a single one already seems kind of questionable.